src/superqt/utils/_throttler.py

```diff
@@ -227,16 +230,46 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> "Future[R]":  # noqa
         return self._future
 
     def _set_future_result(self):
-        result = self.__wrapped__(*self._args[: self._max_args], **self._kwargs)
+        if self._instance is not None:
+            result = self.__wrapped__(
+                self._instance, *self._args[: self._max_args], **self._kwargs
+            )
+        else:
+            result = self.__wrapped__(*self._args[: self._max_args], **self._kwargs)
         self._future.set_result(result)
 
+    def __set_name__(self, owner, name):
+        self._name = name
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return self
+        parent = self.parent()
+        if parent is None and isinstance(instance, QObject):
+            parent = instance
+
+        setattr(
+            instance,
+            self._name,
```

Comment:
type checker is pointing out that self._name could technically be `None`...  maybe raise an exception if we get there? (i've seen cases where `__set_name__` isn't called by various metaclasses... probably very rare, but might as well please the linter)

Reply:
I improve initial if to handle this.